tests/test-customize-changesets-controller.php

```diff
@@ -68,16 +137,211 @@ public function test_get_item_schema() {
 	 * Test get_items.
 	 *
 	 * @covers WP_REST_Customize_Changesets_Controller::get_items()
 	 */
 	public function test_get_items() {
-		$this->markTestIncomplete();
+		wp_set_current_user( self::$admin_id );
+
+		$manager1 = new WP_Customize_Manager();
+		$manager1->save_changeset_post();
+
+		$manager2 = new WP_Customize_Manager();
+		$manager2->save_changeset_post();
+
+		$request = new WP_REST_Request( 'GET', sprintf( '/customize/v1/changesets' ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertNotInstanceOf( 'WP_Error', $response );
+		$response = rest_ensure_response( $response );
+		$this->assertEquals( 200, $response->get_status() );
+		$this->assertEquals( 2, count( $response->get_data() ) );
+	}
+
+	/**
+	 * Add custom settings for testing.
+	 *
+	 * @param object $wp_customize WP Customize Manager.
+	 */
+	public function custom_settings_customze_register( $wp_customize ) {
+		$wp_customize->add_setting( 'allowed_setting' );
+		$wp_customize->add_setting( 'forbidden_setting', array(
+			'capability' => 'do_not_allow',
+		) );
+	}
+
+	/**
+	 * Test the case when user doesn't have permissions for some of the settings.
+	 */
+	public function test_get_item_without_permissions_to_some_settings() {
+		wp_set_current_user( self::$admin_id );
+
+		$setting_allowed_id = 'allowed_setting';
+		$setting_forbidden_id = 'forbidden_setting';
+
+		$manager = new WP_Customize_Manager();
+		$manager->save_changeset_post( array(
+			'status' => 'auto-draft',
+			'data' => array(
+				$setting_allowed_id => array(
+					'value' => 'Foo',
+				),
+				$setting_forbidden_id => array(
+					'value' => 'Bar',
+				),
+			),
+		) );
+
+		add_action( 'customize_register', array( $this, 'custom_settings_customize_register' ) );
+
+		$request = new WP_REST_Request( 'GET', sprintf( '/customize/v1/changesets/%s', $manager->changeset_uuid() ) );
+		$response = $this->server->dispatch( $request );
+
+		$this->assertNotInstanceOf( 'WP_Error', $response );
+		$response = rest_ensure_response( $response );
+
+		$changeset_data = $response->get_data();
+		$this->assertSame( array(), $changeset_data );
+		$changeset_settings = json_decode( $changeset_data['settings'], true );
+
+		$this->assertArrayHasKey( $setting_allowed_id, $changeset_settings );
+		$this->assertFalse( isset( $changeset_settings[ $setting_forbidden_id ] ) );
+
+		remove_action( 'customize_register', array( $this, 'custom_settings_customize_register' ) );
+	}
+
+	/**
+	 * Filter for GET request.
+	 *
+	 * @param object $response Rest response.
+	 * @return object mixed Filtered data.
+	 */
+	public function get_changeset_custom_callback( $response ) {
+
+		$changeset_data = array(
+			'foo' => array(
+				'value' => 'bar',
+			),
+		);
+		$json_options = 0;
+		if ( defined( 'JSON_UNESCAPED_SLASHES' ) ) {
+			$json_options |= JSON_UNESCAPED_SLASHES;
+		}
+		$json_options |= JSON_PRETTY_PRINT;
+		$response->data['settings'] = wp_json_encode( $changeset_data, $json_options );
```

Comment:
Object because it has string keys, mapping the setting ID to yet another object that contains the setting value. In PHP such a construct can be represented as an array or as an object, but in JavaScript it can only be represented as an `object`, like:

```json
{
    "blogname": { "value": "Awesome Website" }
}
```